src/recipe.cpp

```diff
@@ -236,7 +240,11 @@ void recipe::add_requirements( const std::vector<std::pair<requirement_id, int>>
 
 std::string recipe::get_consistency_error() const
 {
-    if( !item::type_is_defined( result_ ) ) {
+    if( !item::type_is_defined( result_ )  && category != "CC_BUILDING") {
```

Comment:
It shouldn't be determined by `category`. `category` is just a description, not type. Instead you should add an `enum` that specifies whether the recipe is an item recipe or building recipe.